src/api/app/services/workflows/yaml_downloader.rb

```diff
@@ -54,5 +54,18 @@ def gitea_download_url
         "#{@scm_payload[:api_endpoint]}/#{@scm_payload[:target_repository_full_name]}/raw/branch/#{@scm_payload[:target_branch]}/#{@token.workflow_configuration_path}"
       end
     end
+
+    # Note: For GitLab we still use the Down gem when workflow_configuration_url is present
+    def download_gitlab_yaml_file
+      gitlab_client = Gitlab.client(endpoint: "#{@scm_payload[:api_endpoint]}/api/v4", private_token: @token.scm_token)
+      gitlab_file = gitlab_client.file_contents(@scm_payload[:project_id], @token.workflow_configuration_path, @scm_payload[:target_branch])
+
+      tempfile = Tempfile.new(["#{Time.zone.now}", '.yaml'])
+      tempfile.write(gitlab_file)
+      tempfile.rewind
+      tempfile
+    rescue Gitlab::Error::NotFound => e
+      raise Token::Errors::NonExistentRepository, e.message
```

Comment:
```suggestion
      begin
        gitlab_client = Gitlab.client(endpoint: "#{@scm_payload[:api_endpoint]}/api/v4", private_token: @token.scm_token)
        gitlab_file = gitlab_client.file_contents(@scm_payload[:project_id], @token.workflow_configuration_path, @scm_payload[:target_branch])
      rescue Gitlab::Error::NotFound => e
        raise Token::Errors::NonExistentRepository, e.message
      end

      tempfile = Tempfile.new(["#{Time.zone.now}", '.yaml'])
      tempfile.write(gitlab_file)
      tempfile.rewind
      tempfile
```
I would be more precise about where to rescue from the Gitlab exception. It would help to understand better where the exception can occur.

Reply:
Okay, It's updated

Reply:
> Okay, It's updated

Hm... not the way I suggested. In fact, I prefer the original that this last version you propose.

To make more clear what I wanted to express in my first comment, I wanted the added `begin` block to rescue only the lines of code where the exception could be raised up.

In other words, it doesn't make much sense to try to rescue a `Gitlab::Error::NotFound` exception from lines where it cannot occur (those which start with `tempfile`). So we can take out from the `begin` block the lines which start with `tempfile`.

Reply:
I understand what you are saying. I reverted the `begin` statement because Rubocop is complaining about that. I'd instead leave it like this, without `begin`, because it's more clear to read

Reply:
> I understand what you are saying. I reverted the `begin` statement because Rubocop is complaining about that.

Sure. That what not what I suggested.

> I'd instead leave it like this, without `begin`, because it's more clear to read

Not enough to convince me...